### ISSUE TITLE
Check for required in subparser only if subparser was invoked

### ISSUE THREAD

**Issue by myOmikron** [hellflame/argparse#11]
Hey again :)

In my opinion it would be a better approach to check for required arguments only in the current context, instead of summing up all required arguments from all parsers. 

Example:
```go
	mainParser := argparse.NewParser("main", "", nil)
	testParser := mainParser.AddCommand("test", "", nil)

	mainParser.String("", "abc", &argparse.Option{})
	testParser.String("", "def", &argparse.Option{Required: true})

	if err := mainParser.Parse(nil); err != nil {
		fmt.Println(err.Error())
		os.Exit(1)
	}
        
        fmt.Println("Program exited normally")
```
Program is executed with the following parameters:
```bash
$ test --abc 1
``````

**Expected behavior**:
Output: `Program exited normally` as `test` was not provided as command.

**Current behavior**:
```bash
$ test --abc 1
DEF is required
```
This will break every use of Required in a subparser.

I'd be happy to create a PR to "fix" this issue.

**Comment by hellflame** [comment]
thanks again :)
this really is a bug that need to fix.

the problem must be caused by these lines
https://github.com/hellflame/argparse/blob/master/parse.go#L505-L507

give me some time to think about why I added these lines 😅

**Comment by myOmikron** [comment]
> the problem must be caused by these lines https://github.com/hellflame/argparse/blob/master/parse.go#L505-L507

I agree, after changing to [this:](https://github.com/myOmikron/argparse/blob/featureRequiredContext/parse.go#L505-L507)
```go
	if subParser != nil {
		entries = append(entries, append(subParser.entries, subParser.positionArgs...)...)
	}
```
everything works as expected.

**Comment by hellflame** [comment]
that's right. 

but there's a test case fail, it reminds me of my pool test case about sub parser. 

let me do the fix and I'll adjust test case a bit by the way, to better test the sub parser and future test cases. 

thanks again :)

**Comment by hellflame** [comment]
fixed & published version: v1.7.3

**Comment by myOmikron** [comment]
Thanks for the fast fix :)
